stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/plugin/listeners/ChannelMarkReadListener.kt

```diff
@@ -0,0 +1,18 @@
+package io.getstream.chat.android.client.experimental.plugin.listeners
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.utils.Result
+
+/**
+ * Listener for [ChatClient.markRead] requests
+ */
+public interface ChannelMarkReadListener {
+    /**
+     * Run precondition for the request. If it returns [Result.isSuccess] then the request is run otherwise it returns
+     * [Result.error] and no request is made.
+     */
```

Comment:
Missing docs for the params here.

It should have the following notation

@param
@param
....
@return (if it has a return value